test/pkg/installers/hyperfoil_node_scraper_installer.go

```diff
@@ -0,0 +1,247 @@
+)
+
+// GetHyperfoilNodeScraperInstaller returns Hyperfoil Node scraper installer
+func GetHyperfoilNodeScraperInstaller() ServiceInstaller {
+	return &hyperfoilNodeScraperYamlNamespacedInstaller
+}
+
+func installHyperfoilNodeScraper(namespace string) error {
+	framework.GetLogger(namespace).Info("Deploy Hyperfoil Node scraper")
+
+	if err := framework.CreateServiceAccount(namespace, getHyperfoilNodeScraperUniqueName(namespace)); err != nil {
+		return err
+	}
+	if err := createHyperfoilNodeScraperClusterRole(getHyperfoilNodeScraperUniqueName(namespace)); err != nil {
+		return err
+	}
+	if err := createHyperfoilNodeScraperClusterRoleBinding(getHyperfoilNodeScraperUniqueName(namespace), namespace); err != nil {
+		return err
+	}
+
+	scraperConfig, err := getNodeScraperConfigContent(namespace)
+	if err != nil {
+		return err
+	}
+
+	err = framework.CreateConfigMap(namespace, NodeScraperStartConfigMapName, map[string]string{".node-scraper-config.json": scraperConfig, "99-node-scraper-start.sh": nodeScraperStart}, nil)
+	if err != nil {
+		return err
+	}
+	err = framework.CreateConfigMap(namespace, NodeScraperStopConfigMapName, map[string]string{"00-node-scraper-stop.sh": nodeScraperStop}, nil)
+	if err != nil {
+		return err
+	}
+
+	return deployNodeScraper(namespace)
+}
+
+func waitForHyperfoilNodeScraperRunning(namespace string) error {
+	return framework.WaitForPodsWithLabel(namespace, "app", hyperfoilNodeScraperDeploymentName, 1, 3)
+}
+
+func uninstallHyperfoilNodeScraper(namespace string) error {
+	var originalError error
+
+	// Delete cluster wide resources, the rest is deleted together with namespace
+	crb, err := framework.GetClusterRoleBinding(getHyperfoilNodeScraperUniqueName(namespace))
+	if err != nil {
+		framework.GetLogger(namespace).Error(err, fmt.Sprintf("Cannot retrieve ClusterRoleBinding %s", getHyperfoilNodeScraperUniqueName(namespace)))
+		if originalError == nil {
+			originalError = err
+		}
+	} else {
+		if err = framework.DeleteObject(crb); err != nil {
+			framework.GetLogger(namespace).Error(err, fmt.Sprintf("Cannot delete ClusterRoleBinding %s", getHyperfoilNodeScraperUniqueName(namespace)))
+			if originalError == nil {
+				originalError = err
+			}
+		}
+	}
+
+	cr, err := framework.GetClusterRole(getHyperfoilNodeScraperUniqueName(namespace))
+	if err != nil {
+		framework.GetLogger(namespace).Error(err, fmt.Sprintf("Cannot retrieve ClusterRole %s", getHyperfoilNodeScraperUniqueName(namespace)))
+		if originalError == nil {
+			originalError = err
+		}
+	} else {
+		if err = framework.DeleteObject(cr); err != nil {
+			framework.GetLogger(namespace).Error(err, fmt.Sprintf("Cannot delete ClusterRole %s", getHyperfoilNodeScraperUniqueName(namespace)))
+			if originalError == nil {
+				originalError = err
+			}
+		}
+	}
+
+	return originalError
+}
+
+func getHyperfoilNodeScraperCrsInNamespace(namespace string) ([]kubernetes.ResourceObject, error) {
+	return []kubernetes.ResourceObject{}, nil
```

Comment:
No cleanup needed here, so no lookup is being made?

Reply:
Node scraper is not an operator, so it doesn't have any CustomResource.
Ideally I could adjust the generic Installer framework logic as currently the YAML installer is meant to install operators using YAML, not a generic service. So this is a small workaround to avoid larger modifications.